test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfos.java

```diff
@@ -0,0 +1,134 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.queries.HapiQueryOp;
+import com.hedera.services.bdd.spec.transactions.TxnUtils;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.NftID;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.TokenGetNftInfosQuery;
+import com.hederahashgraph.api.proto.java.TokenNftInfo;
+import com.hederahashgraph.api.proto.java.Transaction;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.Assert;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.hedera.services.bdd.spec.queries.QueryUtils.answerCostHeader;
+import static com.hedera.services.bdd.spec.queries.QueryUtils.answerHeader;
+
+public class HapiGetTokenNftInfos extends HapiQueryOp<HapiGetTokenNftInfos> {
+
+	private static final Logger log = LogManager.getLogger(HapiGetTokenNftInfos.class);
+
+	private String token;
+
+	private long start;
+
+	private long end;
+
+	private Optional<List<HapiTokenNftInfo>> expectedNfts = Optional.empty();
+
+	public HapiGetTokenNftInfos(String token, long start, long end) {
+		this.token = token;
+		this.start = start;
+		this.end = end;
+	}
+
+	@Override
+	public HederaFunctionality type() {
+		return HederaFunctionality.TokenGetNftInfos;
+	}
+
+	@Override
+	protected void assertExpectationsGiven(HapiApiSpec spec) throws Throwable {
+		var actualInfo = response.getTokenGetNftInfos().getNftsList();
+		expectedNfts.ifPresent(nfts -> {
+			for (HapiTokenNftInfo nftInfo : nfts) {
+				var expectedNftId = NftID.newBuilder();
+				var expectedNftElement = TokenNftInfo.newBuilder();
+
+				nftInfo.getExpectedTokenID().ifPresent(e -> {
+					expectedNftId.setTokenID(TxnUtils.asTokenId(e, spec));
+					expectedNftElement.setCreationTime(spec.registry().getCreationTime(e));
+				});
+				nftInfo.getExpectedSerialNum().ifPresent(expectedNftId::setSerialNumber);
+
+				expectedNftElement.setNftID(expectedNftId.build());
+				nftInfo.getExpectedAccountID().ifPresent(e -> expectedNftElement.setAccountID(TxnUtils.asId(e, spec)));
+				nftInfo.getExpectedMetadata().ifPresent(expectedNftElement::setMetadata);
+
+				var completedNft = expectedNftElement.build();
+				Assert.assertTrue(actualInfo.contains(completedNft));
```

Comment:
There is a bug in this implementation. See https://github.com/hashgraph/hedera-services/pull/1724/files#diff-51458346123a0079d1ac75c581029bf8ea203fffcb45dc77eb24f987b964933aR91 for reference. 

https://github.com/hashgraph/hedera-services/pull/1615#discussion_r662168616